src/ClientImpl.cpp

```diff
@@ -575,9 +694,30 @@ void ClientImpl::createConnection(const std::string& hostname,
         }
     }
     if (keepConnecting) {
+        if (pc->m_bufferEvent != NULL)  {
+            bufferevent_free(pc->m_bufferEvent);
+            pc->m_bufferEvent = NULL;
+        }
         createPendingConnection(hostname, port);
     } else {
-        throw ConnectException();
+        // if no error has been reported for the connection, back off and listen if
+        // any events were there to process before calling it no-connection
+        int retry = 0;
+        while (pc->m_status && retry < 5) {
```

Comment:
This is to workaround the timing scenario I was running into where the connection to the server is seen by client but client has not been received authentication response for the request it has sent. So for that case, it can flag it as no-connection even though server is out there and it's just a delayed response. This happens sometimes running multiple server nodes in VM.